src/main/scala/Color.scala

```diff
@@ -1,17 +1,21 @@
 package chess
 
-sealed trait Color:
+import scala.annotation.targetName
 
-  inline def fold[A](w: => A, b: => A): A = if (white) w else b
+enum Color:
+  case White, Black
 
-  def unary_! : Color
+  def fold[A](w: => A, b: => A): A = if (white) w else b
 
-  val passablePawnRank: Rank
-  val promotablePawnRank: Rank
-  val backRank: Rank
+  @targetName("negate")
```

Comment:
I'm not sure this is necessary. probably not. It is only useful for other JVM language not Scala.

https://docs.scala-lang.org/scala3/reference/other-new-features/targetName.html